modules/gapi/src/backends/fluid/gfluidcore_simd_sse42.hpp

```diff
@@ -0,0 +1,576 @@
+                                      const short beta) {
+    constexpr int nlanes = 16;
+    __m128i zero = _mm_setzero_si128();
+    __m128i b = _mm_set1_epi16(beta);
+
+    for (int w = 0; inLength >= nlanes;)
+    {
+        for (; w <= inLength - nlanes; w += nlanes)
+        {
+            __m128i s0 = _mm_lddqu_si128(reinterpret_cast<const __m128i*>(&src0[w]));
+            __m128i s1 = _mm_lddqu_si128(reinterpret_cast<const __m128i*>(&src1[w]));
+            __m128i a1 = _mm_unpacklo_epi8(s0, zero);
+            __m128i b1 = _mm_unpacklo_epi8(s1, zero);
+            __m128i a2 = _mm_unpackhi_epi8(s0, zero);
+            __m128i b2 = _mm_unpackhi_epi8(s1, zero);
+            __m128i r1 = _mm_mulhrs_epi16(_mm_sub_epi16(a1, b1), b);
+            __m128i r2 = _mm_mulhrs_epi16(_mm_sub_epi16(a2, b2), b);
+            __m128i res1 = _mm_add_epi16(r1, b1);
+            __m128i res2 = _mm_add_epi16(r2, b2);
+            _mm_storeu_si128(reinterpret_cast<__m128i*>(tmp + w), _mm_packus_epi16(res1, res2));
+        }
+
+        if (w < inLength) {
+            w = inLength - nlanes;
+            continue;
+        }
+        break;
+    }
+}
+
+template<int chanNum>
+CV_ALWAYS_INLINE void horizontal_anyLPI(uint8_t* dst,
+                                        const uchar* src, const short mapsx[],
+                                        const short alpha[], const int width)
+{
+    constexpr int nlanes = 16;
+    __m128i zero = _mm_setzero_si128();
+
+    for (int x = 0; width >= nlanes;)
+    {
+        for (; x <= width - nlanes; x += nlanes)
+        {
+            __m128i a012 = _mm_setr_epi16(alpha[x], alpha[x], alpha[x], alpha[x + 1],
+                                          alpha[x + 1], alpha[x + 1], alpha[x + 2], alpha[x + 2]);
+            __m128i a2345 = _mm_setr_epi16(alpha[x + 2], alpha[x + 3], alpha[x + 3], alpha[x + 3],
+                                           alpha[x + 4], alpha[x + 4], alpha[x + 4], alpha[x + 5]);
+
+            __m128i a567 = _mm_setr_epi16(alpha[x + 5], alpha[x + 5], alpha[x + 6], alpha[x + 6],
+                                          alpha[x + 6], alpha[x + 7], alpha[x + 7], alpha[x + 7]);
+            __m128i a8910 = _mm_setr_epi16(alpha[x + 8], alpha[x + 8], alpha[x + 8], alpha[x + 9],
+                                           alpha[x + 9], alpha[x + 9], alpha[x + 10], alpha[x + 10]);
+
+            __m128i a10111213 = _mm_setr_epi16(alpha[x + 10], alpha[x + 11], alpha[x + 11], alpha[x + 11],
+                                               alpha[x + 12], alpha[x + 12], alpha[x + 12], alpha[x + 13]);
+            __m128i a131415 = _mm_setr_epi16(alpha[x + 13], alpha[x + 13], alpha[x + 14], alpha[x + 14],
+                                             alpha[x + 14], alpha[x + 15], alpha[x + 15], alpha[x + 15]);
+
+            __m128i a1 = _mm_setr_epi8(src[chanNum * (mapsx[x] + 0)],     src[chanNum * (mapsx[x] + 0) + 1],     src[chanNum * (mapsx[x] + 0) + 2],
```

Comment:
Overall structure suppose that chanNum is equall to 3. Looks like instance with chanNum == 4 will lose last channel. Is templating necessary here?